### ISSUE TITLE
Adapt disabled test checking string.format("%q", ...) from PUC-Rio Lua 5.1 test suite in strings.lua

### ISSUE THREAD

**Issue by Buristan** [tarantool/tarantool#5710]
This ticket is a logical continuation of gh-4473.

If run  PUC-Rio Lua 5.1 test suite by LuaJIT (HEAD at the moment: [`1672bdc`](https://github.com/tarantool/luajit/commit/1672bdc0ffbee9f32fadb6943909d6af7eb9a7b1)) (not Tarantool) it fails with the following error:
```
/home/burii/builds_workspace/luajit/gh-4473-adopt-luavela-test-suites/src/luajit: strings.lua:105: assertion failed!
stack traceback:
        [C]: in function 'assert'
        strings.lua:105: in function 'dofile'
        ...adopt-luavela-test-suites/test/PUC-Lua-5.1-tests/all.lua:84: in main chunk
        [C]: at 0x560d03ad1f96
```
caused by this test chunk (assuming that all previous failing test chunks non relating to this error are disabled):
```lua
assert(string.format('%q', "\0") == [["\000"]])
```

LuaJIT since v2.0.0-beta6 has extension from Lua 5.2: `string.format()`: `%q` reversible.
According [this section](https://www.lua.org/manual/5.1/manual.html#pdf-string.format) in Lua 5.1 Reference manual `string.format()` does not accept string values containing embedded zeros, except as arguments to the q option.
In Lua 5.2 '\0' is not handled differently from other control chars in `string.format('%q', ...)`.
See commit 7cc981c14067d4b0e774a6bfb0acfc2f5c911f0d (string.format("%q", str) is now fully reversible (from Lua 5.2).).

**Comment by Buristan** [comment]
This test chunk can be adapted from PUC-Rio Lua 5.2 test suite like this:
```diff
diff --git a/test/PUC-Lua-5.1-tests/strings.lua b/test/PUC-Lua-5.1-tests/strings.lua
index 4f87143..47c0893 100644
--- a/test/PUC-Lua-5.1-tests/strings.lua
+++ b/test/PUC-Lua-5.1-tests/strings.lua
@@ -112,8 +112,8 @@ assert(string.format('%q%s', x, x) == '"\\"
 -- (string.format("%q", str) is now fully reversible
 -- (from Lua 5.2).).
 -- See also https://github.com/tarantool/tarantool/issues/5710.
--- Test is disabled for LuaJIT for now.
--- assert(string.format('%q', "\0") == [["\000"]])
+-- Test is adapted from PUC-Rio Lua 5.2 test suite.
+assert(string.format('%q', "\0") == [["\0"]])
 assert(string.format("\0%c\0%c%x\0", string.byte("á"), string.byte("b"), 140) ==
               "\0á\0b8c\0")
 assert(string.format('') == "")
```

**Comment by igormunkin** [comment]
The issue is resolved in scope of tarantool/luajit@58be0c4. The submodule was updated in tarantool in 2.8.0-249-g8a09e18ce (8a09e18), 2.7.1-210-g5f4298b6f (5f4298b), 2.6.2-203-g8343a8129 (8343a81), 1.10.9-119-g7a4c7d0b0 (7a4c7d0).
